src/main/scala/is/hail/methods/LogisticRegressionBurden.scala

```diff
@@ -0,0 +1,97 @@
+package is.hail.methods
+
+import breeze.linalg._
+import is.hail.expr._
+import is.hail.keytable.KeyTable
+import is.hail.stats._
+import is.hail.utils._
+import is.hail.variant._
+import org.apache.spark.sql.Row
+
+object LogisticRegressionBurden {
+
+  def apply(vds: VariantDataset,
+    keyName: String,
+    variantKeys: String,
+    singleKey: Boolean,
+    aggExpr: String,
+    test: String,
+    yExpr: String,
+    covExpr: Array[String]): (KeyTable, KeyTable) = {
+
+    def tests = Map("wald" -> WaldTest, "lrt" -> LikelihoodRatioTest, "score" -> ScoreTest, "firth" -> FirthTest)
+    if (!tests.isDefinedAt(test))
+      fatal(s"Supported tests are ${ tests.keys.mkString(", ") }, got: $test")
+
+    val logRegTest = tests(test)
+
+    val (y, cov, completeSamples) = RegressionUtils.getPhenoCovCompleteSamples(vds, yExpr, covExpr)
+    val completeSamplesSet = completeSamples.toSet
+    if (completeSamplesSet(keyName))
+      fatal(s"Key name '$keyName' clashes with a sample name")
+
+    val logregFields = logRegTest.schema.asInstanceOf[TStruct].fields.map(_.name).toSet
+    if (logregFields(keyName))
+      fatal(s"Key name '$keyName' clashes with reserved $test logreg columns $logregFields")
+
+    if (!y.forall(yi => yi == 0d || yi == 1d))
+      fatal(s"For logistic regression, phenotype must be Boolean or numeric with all present values equal to 0 or 1")
+
+    val n = y.size
+    val k = cov.cols
+    val d = n - k - 1
+
+    if (d < 1)
+      fatal(s"$n samples and $k ${ plural(k, "covariate") } including intercept implies $d degrees of freedom.")
+
+    val nullModel = new LogisticRegressionModel(cov, y)
+    val nullFit = nullModel.fit()
+
+    if (!nullFit.converged)
+      fatal("Failed to fit (unregulatized) logistic regression null model (covariates only): " + (
```

Comment:
As opposed to Firth null model which is re-fit per variant / key and implicitly regularized.  I've simplified the language.